worlds/_bizhawk/README.md

```diff
@@ -0,0 +1,277 @@
+`watcher_timeout` (the amount of time in seconds between iterations of the game watcher loop).
+
+### Example
+
+A very simple client might look like this. All addresses here are made up, you should obviously be using addresses that
+make sense for your specific ROM. The `validate_rom` here tries to read the name of the ROM. If it gets the value it
+wanted, it sets a couple values on `ctx` and returns `True`. The `game_watcher` reads some data from memory and acts on
+it by sending messages to AP. You should be smarter than this example, which will send `LocationChecks` messages even if
+there's nothing new since the last loop.
+
+```py
+from typing import TYPE_CHECKING
+
+from NetUtils import ClientStatus
+
+import worlds._bizhawk as bizhawk
+from worlds._bizhawk.client import BizHawkClient
+
+if TYPE_CHECKING:
+    from worlds._bizhawk.context import BizHawkClientContext
+
+
+class MyGameClient(BizHawkClient):
+    game = "My Game"
+    system = "GBA"
+    patch_suffix = ".apextension"
+
+    async def validate_rom(self, ctx: "BizHawkClientContext") -> bool:
+        try:
+            # Check ROM name/patch version
+            rom_name = ((await bizhawk.read(ctx.bizhawk_ctx, [(0x100, 6, "ROM")]))[0]).decode("ascii")
+            if rom_name != "MYGAME":
+                return False  # Not a MYGAME ROM
+        except bizhawk.RequestFailedError:
+            return False  # Not able to get a response, say no for now
+
+        # This is a MYGAME ROM
+        ctx.game = self.game
+        ctx.items_handling = 0b001
+        ctx.want_slot_data = True
+
+        return True
+
+    async def game_watcher(self, ctx: "BizHawkClientContext") -> None:
+        try:
+            # Read save data
+            save_data = await bizhawk.read(
+                ctx.bizhawk_ctx,
+                [(0x3000100, 20, "System Bus")]
+            )[0]
+
+            # Check locations
+            if save_data[2] & 0x04:
+                await ctx.send_msgs([{
+                    "cmd": "LocationChecks",
+                    "locations": [23]
+                }])
+
+            # Send game clear
+            if not ctx.finished_game and (save_data[5] & 0x01):
+                await ctx.send_msgs([{
+                    "cmd": "StatusUpdate",
+                    "status": ClientStatus.CLIENT_GOAL
+                }])
+
+        except bizhawk.RequestFailedError:
+            # The connector didn't respond. Exit handler and return to main loop to reconnect
+            pass
+```
+
+### Tips
+
+- Make sure your client gets imported when your world is imported. You probably don't need to actually use anything in
+your `client.py` elsewhere, but you still have to import the file for your client to register itself.
+- When it comes to performance, there are two directions to optimize:  
+  1. When you have to do something all on the same frame, do as little as possible. Only read and write necessary data,
```

Comment:
Shouldn't something be plural here?

Reply:
"Somethings" either isn't a word or is only used if you want the meaning of the phrase/sentence to strongly emphasize the plurality. "Some things" reads much more ambiguously, strangely. Like saying "do the stuff". It sounds lazy, maybe? Hard to explain.

The "something" here is referring to a singular conceptual action like "give an item", or "kill the player". That action may involve multiple steps (in the context of this bullet point, it probably does, otherwise the sentence wouldn't be relevant). What I'm trying to say is, when there is one "thing" you have to do which involves multiple commands that must run on the same frame, do as few commands as possible. "All" meaning more "the entirety of" than "every one of multiple". The bridge is all iced over. The dog ate the treat all at once. Etc...

Way more of an explanation than this probably deserves, lol, but you phrased it as a question. If this wording is confusing or needs clarification, I'm not opposed to changing it. But it sounds fine to my ears, and making it plural doesn't mean anything obvious to me, since the two versions I mentioned above change the perceived meaning so much.

Reply:
```suggestion
  1. If you need to execute multiple commands on the same frame, do as little work as possible. Only read and write necessary data,
```

Is this along the lines of what you are looking for?

Reply:
Sounds fine. I suppose I was avoiding "commands" because for the most part individual commands are abstracted away. But probably not a big deal.